src/components/datepicker/datePicker.js

```diff
@@ -331,7 +331,11 @@
     var parsedDate = this.dateLocale.parseDate(inputString);
     this.dateUtil.setDateTimeToMidnight(parsedDate);
 
-    if (this.dateUtil.isValidDate(parsedDate) &&
+    if (inputString === '') {
```

Comment:
I see this as being very clear from just the code; "if the input is empty, the model is set to null, which is a valid value."